src/en/darthsdroids/src/eu/kanade/tachiyomi/extension/en/darthsdroids/DarthsDroids.kt

```diff
@@ -0,0 +1,222 @@
+        }
+        initialized = true
+    }
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> {
+        val mainArchive = client.newCall(GET(mainArchiveUrl, headers)).execute().asJsoup()
+        val archiveData = mainArchive.select("div.text > table.text > tbody > tr")
+
+        val mangas = mutableListOf<SManga>()
+        var nextMangaTitle = name
+        var nthManga = 0
+
+        run stop@{
+            archiveData.forEach {
+                val maybeTitle = it.selectFirst("th")?.text()?.trim()
+                if (maybeTitle != null) {
+                    nextMangaTitle = "$name $maybeTitle"
+                } else {
+                    val maybeArchive = it.selectFirst("""td[colspan="3"] > a""")?.attr("href")
+                    if (maybeArchive != null) {
+                        mangas.add(dndManga(maybeArchive, nextMangaTitle, SManga.COMPLETED, nthManga))
+                        nthManga += 1
+                    } else {
+                        // We reached the end, assuming the page layout stays consistent beyond D&D8.
+                        // Thus, we append our final manga with this current page as its archive.
+                        // Unfortunately this means we will needlessly fetch this page twice.
+                        mangas.add(dndManga("/archive.html", nextMangaTitle, SManga.ONGOING, nthManga))
+                        return@stop
+                    }
+                }
+            }
+        }
+
+        return Observable.just(MangasPage(mangas, false))
+    }
+
+    // Not efficient, but the simplest way for me to refresh
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        fetchPopularManga(0)
+            .map { mangasPage ->
+                mangasPage
+                    .mangas
+                    .first { it.url == manga.url }!!
+            }
+
+    // This implementation here is needlessly complicated, for it has to automatically detect
+    // whether we’re in a date-annotated archive, the main archive, or a dateless archive.
+    // All three are largely similar, there are just *some* (annoying) differences we have to
+    // deal with.
+    private fun fetchChaptersForDatedAndDatelessArchives(archivePages: Document, archivePath: String): List<SChapter> {
+        val chapters = mutableListOf<SChapter>()
+        var i = 0
+        archivePages.select("""div.text > table.text > tbody > tr""").forEach {
+            val pageData = it.select("""td""")
+            var pageAnchor = pageData.getOrNull(2)?.selectFirst("a")
+            // null for »Intermission«, main archive, dateless archive,…
+            if (pageAnchor != null) {
+                val a = pageAnchor
+
+                chapters.add(
+                    SChapter.create().apply {
+                        name = a.text()
+                        chapter_number = i.toFloat()
+                        date_upload = DATE_FMT.parse(pageData[0].text().trim())?.time ?: 0L
```

Comment:
Wrap in `runCatching`

Reply:
Sure. Can you point me at some documentation? I don’t know what that method does or which type implements it.

Reply:
On mobile, so responses are short.

https://github.com/search?q=repo%3Akeiyoushi%2Fextensions-source%20date_upload%20%3D%20runcatching&type=code